### DISCUSSION TITLE
When print, use different colors for the key and value of the dict

### DISCUSSION THREAD

**Discussion by GeekDuanLian** [Textualize/rich#1042] [Ideas] [answered]
It seems that the current key and value are both in green. If the data is dense, it may be dazzling.
if can consider changing the key to other colors, such as `jq` using blue.

**Comment by willmcgugan** [comment, marked as answer]
It depends entirely on your terminal. Looks fine with a well chosen palette. Windows has the most awful defaults. See the docs on Themes if you want to change the syntax highlighting.

**Reply by GeekDuanLian** [reply]
Okay, let me read the documentation carefully, there are too many features.

**Reply by GeekDuanLian** [reply]
I read the document and found that there seems to be no setting about dict, only the setting based on data type through repr.

**Reply by willmcgugan** [reply]
Try this:

```python
from rich.console import Console
from rich.theme import Theme

my_theme = Theme({"repr.str": "blue"})
console = Console(theme=my_theme)
console.print({"hello": "world"})
```

**Reply by GeekDuanLian** [reply]
What I want to set is the different colors of key and value.
For example, the key is blue but the value is still green.
In your example, the key and value are still the same color

**Reply by willmcgugan** [reply]
There's currently no way to do that, I'm afraid. You could do it by creating a customized highlighter. Set a different style for a value followed by a colon.

**Reply by jiamo** [reply]
Thanks. I came out a solution:

theme = Theme(
    {
        "repr.str": "bright_blue",
        "repr.value_str": "green",
    }
)

a highlight like 

r":(?<![\\\w]) (?P<value_str>b?\'\'\'.*?(?<!\\)\'\'\'|b?\'.*?(?<!\\)\'|b?\"\"\".*?(?<!\\)\"\"\"|b?\".*?(?<!\\)\")"

**Reply by GeekDuanLian** [reply]
> Thanks. I came out a solution:
> 
> theme = Theme(
> {
> "repr.str": "bright_blue",
> "repr.value_str": "green",
> }
> )
> 
> a highlight like
> 
> r":(?<![\\w]) (?P<value_str>b?'''._?(?<!\)'''|b?'._?(?<!\)'|b?"""._?(?<!\)"""|b?"._?(?<!\)")"

Where should regex be entered? Can you give the full code?

**Reply by jiamo** [reply]
can you have  a look at https://stackoverflow.com/questions/26459749/pretty-printing-json-with-ascii-color-in-python/68190273#68190273

**Reply by GeekDuanLian** [reply]
> can you have a look at https://stackoverflow.com/questions/26459749/pretty-printing-json-with-ascii-color-in-python/68190273#68190273

Looks great! I hope the author can incorporate this feature.
